### PR TITLE
Redesign home page

### LINKED ISSUE #475: Re-design landing page
In addition to the current content, things to cover:

- How to deploy for free: use Cloud Run: https://google.github.io/mesop/guides/deployment/
- Web components
- Auth guide
- How is this different from other frameworks? https://google.github.io/mesop/blog/2024/05/13/why-mesop/
- Address: not an official Google product - used by dozens of teams at Google to build demos and internal apps.

Feature YT videos like https://www.youtube.com/watch?v=Zi1DnGOrYho&t=1s
Feature tweets

### PR DESCRIPTION
**Pull request by wwwillchen**
Fixes #475

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -3,85 +3,326 @@ hide:
...
+=== "Streaming UI"
 
-__Step 1:__ Install it
+    ```python
+    --8<-- "mesop/examples/docs/streaming.py"
+    ```
+
+=== "Composable"
+
+    ``` python
```
Comment (richard-to) [inline docs/index.md:105]: I like showing how Mesop is Composable, but this example is not the clearest to me out of context. Maybe it's the use of "scaffolding"?

File: docs/index.md [hunk 2]
```diff
@@ -3,85 +3,326 @@ hide:
...
+        me.slot()
+
+    def page1():
+      with scaffold(url="/page1"):
+        some_content(...)
+    ```
+
+
+<section id="modern-ui-principles" class="container blue-gradient-background" style="padding-bottom: var(--modern-ui-principles-offset)">
+  <div class="section-chip blue">Modern UI principles</div>
```
Comment (richard-to) [inline docs/index.md:119]: I noticed some chips are bold and some are not. This one is not bold. I think the bold works better.

File: docs/index.md [hunk 3]
```diff
@@ -3,85 +3,326 @@ hide:
...
+  <div class="section-headline">
+    Declarative UI that's easy to understand
+  </div>
+  <div class="section-body">
+    Mesop embraces a declarative approach to UI development, making your code more intuitive and maintainable. Say goodbye to complex imperative logic and hello to clean, expressive code. Build user interaces using battle-tested patterns, without leaving Python.
```
Comment (richard-to) [inline docs/index.md:124]: [nit] The sentences here seem a bit disjoint. Doesn't feel like they flow together.

File: docs/index.md [hunk 4]
```diff
@@ -3,85 +3,326 @@ hide:
...
+    state = me.state(State)
+    state.image_data = base64.b64encode(e.file.read()).decode()
+    state.detections = detect_objects(e.file)
 ```
 
-## Next Steps
+<section class="container pink-gradient-background">
+  <div class="section-chip pink">Building blocks</div>
+  <div class="section-headline">
+    Ready-to-use components
```
Comment (richard-to) [inline docs/index.md:162]: [nit] Could use a snappier tag line here.

File: docs/index.md [hunk 5]
```diff
@@ -3,85 +3,326 @@ hide:
...
+  </div>
+  <div class="section-body">
+    Mesop provides a versatile range of <a href="./components/">30 components</a>, from low-level building blocks to high-level, AI-focused components. This flexibility lets you rapidly prototype ML apps or build custom UIs, all within a single framework that adapts to your project's use case.
+
+    <a class="section-button" href="./components/chat">
```
Comment (richard-to) [inline docs/index.md:167]: This part seemed weird. Was there supposed to be a demo or image to the right?

File: docs/index.md [hunk 6]
```diff
@@ -3,85 +3,326 @@ hide:
...
+    </a>
+    <a class="section-button" href="./components/table">
+      Data display components
+      <div class="section-button-subtext">e,g. table, plot</div>
+    </a>
+  </div>
+</section>
+
+<section class="container teal-gradient-background">
+  <div class="section-chip teal">Build anything</div>
```
Comment (richard-to) [inline docs/index.md:183]: lol, glad you kept Build anything

File: docs/index.md [hunk 7]
```diff
@@ -3,85 +3,326 @@ hide:
...
+  <div class="section-headline">
+    Build any user interface you can imagine
+  </div>
+  <div class="section-body">
+With Mesop, you can build virtually any web-based user interface or application you can imagine. From quick prototypes to enterprise tools, Mesop provides the customizability to bring your ideas to life.
+  </div>
+  <div class="center">
```
Comment (richard-to) [inline docs/index.md:190]: I'm not a fan of the centering of the iframe against the left alignment of the content. But that is more of a pet peeve of mine. So I think fine as is? How come the iframe doesn't just fit the full width of the section?

File: docs/index.md [hunk 8]
```diff
@@ -3,85 +3,326 @@ hide:
...
+  </div>
+</section>
+
+<section class="container amber-gradient-background">
+  <div class="section-chip amber">Extensible</div>
+  <div class="section-headline">
+  Seamlessly integrate JS with web components
+  </div>
+  <div class="section-body">
+With Mesop <a href="./web_components/">web components</a>, you get the best of both worlds: the simplicity and power of Python, with the ability to drop down into JavaScript when you need that extra control.  Use existing JS libraries - all while keeping your core logic in Python.
```
Comment (richard-to) [inline docs/index.md:202]: The second sentence feels a bit disjoint. Second second sentence feels like tag line that didn't chosen as a tag line. But was still important to keep in.

File: docs/index.md [hunk 9]
```diff
@@ -3,85 +3,326 @@ hide:
...
+</section>
+
+<section class="container blue-gradient-background">
+  <div class="section-chip blue">Simple deployment</div>
+  <div class="section-headline">
+    Deploy your app and share in minutes
+  </div>
+  <div class="section-body">
+Mesop streamlines cloud deployment, enabling you to share your AI application with the world in minutes. With step-by-step guides for deploying to Google Cloud Run or any cloud service that takes a container, you can go from local development to production-ready deployment without wrestling with complex server setups.
+<a class="section-button" href="./guides/deployment" style="display: flex">
```
Comment (richard-to) [inline docs/index.md:213]: Should this go directly to Cloud Run section?

File: docs/index.md [hunk 10]
```diff
@@ -3,85 +3,326 @@ hide:
...
+    Delightful developer experience
+  </div>
+  <div class="section-body">
+Mesop streamlines app development with features like hot reload and strong IDE support with static types, eliminating friction and boosting productivity.
+</div>
+  <div class="dev-gif-row">
+    <div class="gif-caption">
+      Instant hot reload
+    </div>
+    <img src="./assets/hot_reload.gif" class="dev-gif box-shadow">
```
Comment (richard-to) [inline docs/index.md:235]: [nit] Minor thing, but maybe record an example without a bunch of tabs open. Though I guess that's more realistic.

File: docs/index.md [hunk 11]
```diff
@@ -3,85 +3,326 @@ hide:
...
+
+<section class="container teal-gradient-background">
+  <div class="section-chip teal">Community</div>
+  <div class="section-headline">
+    See what others are saying
+  </div>
+  <div class="section-body">
+ Join developers around the world who are building AI apps in Mesop.
+  </div>
+    <div class="center">
```
Comment (richard-to) [inline docs/index.md:253]: [nit] Again, not sure about the center alignment here. But I think not an issue once we get more things to put here. :)

**Comment by wwwillchen** [comment]
> Overall, looks much, much better than the current landing page. Really great work. I like the dark theme. Light theme is good. The little colorations you added to each section were a nice touch. Wonder where you got that idea from lol.
> 
> I do think there's some improvements that can be made. Some sections feel a bit bare, like the deployment section where all we have is Cloud Run button. Or the Extensible section where there's no example yet (but I think that's just because the demo doesn't support the web components yet). Also had some other minor nitpicks, but mostly opinion.

Thanks for brainstorming and the detailed review!

**Pull request merged**